src/journal/journald.c

```diff
@@ -25,8 +25,8 @@ int main(int argc, char *argv[]) {
 
         namespace = argc > 1 ? empty_to_null(argv[1]) : NULL;
 
-        log_set_prohibit_ipc(true);
-        log_set_target(LOG_TARGET_AUTO);
+        log_set_prohibit_ipc(true); /* better safe than sorry */
+        log_set_target(LOG_TARGET_KMSG);
```

Comment:
could you please add some more details on how this and https://github.com/systemd/systemd/commit/bc694c06e60505efeb09e5278a7b22cdfa23975e are related? I'm not familiar with the journal, and after a minute looking at both diffs still it's not clear to me

Reply:
Sure, with bc694c06e60505efeb09e5278a7b22cdfa23975e, journald was no more considered as a system service (`stderr_is_journal()` returns false) and therefore kmsg was no more considered as a possible target. In this case only the console was (when the target is set to `LOG_TARGET_AUTO`).

Reply:
I updated the commit message to explain this better.

Reply:
Thanks - so for my understanding (not that familiar with the journal side of things), before the change was KMSG the only target? Or was the console also used?

Reply:
see log_open(), the console is always used as fallback.

Reply:
I see, thanks

Reply:
Hmm, shouldn't the log target depend on `namespace` being set? If we're in a namespace, logging to the main daemon should be OK. Also, when starting `journald` for testing from the console, logging to the console should be OK too...

Reply:
hmm, interesting i never thought about the namespaced instances to log to the main one. Does that work? Is it really desirable though? that way you link them up to each other after all?

I'd probably say that kmsg is always the way to go, it's too complicated to even explain that instances can prpoagate stuff to the main instance, and I don't see the major benefit, given that journald itself logs so little, and via kmsg it does end up in the main instance anyway.

I do agree with the console comment though. I do invoke journald from the cmdline for testing, and i think it is worth if it logs to the tty then

Reply:
But /dev/kmsg isn't available in containers, is it ?

Reply:
so we would need to do deal with the console case specifically : `log_set_target(isatty(STDERR_FILENO) > 0 ? LOG_TARGET_CONSOLE : LOG_TARGET_KMSG)` ?